services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java

```diff
@@ -314,7 +314,10 @@ private void buildPredicates(RuleFilter filter, List<String> suffixes, List<Stri
       switch (dateFilterType) {
       case STARTED:
         suffixes.add("StartDateAndEndDate");
-        predicates.add("r.startDate <= :date AND r.endDate >= :date");
+        predicates.add("((r.startDate <= :date AND r.endDate >= :date) OR" +
+                       " (r.startDate IS NULL AND r.endDate IS NULL) OR" +
+                       " (r.startDate IS NULL AND r.endDate >= :date) OR" +
+                       " (r.endDate IS NULL AND r.startDate <= :date))");
```

Comment:
Is it similar to this suggestion ? I think it's more clear for code readability (same for other predicates)

```suggestion
        predicates.add("((r.startDate IS NULL OR r.startDate <= :date)"
                       "  AND (r.endDate IS NULL OR r.endDate >= :date))");
```

Reply:
Done!

Reply:
Seems `r.type` is missing, see 
https://github.com/Meeds-io/gamification/compare/55ffc1fa5104f0cb7e29314c62e2a47027f52bce..42ee2d4fb3f979c895c7f070be5f9bdcd831f126

Reply:
ok, seems useless after this change.